ts-tests/containers/MetaDContainer.ts

```diff
@@ -0,0 +1,166 @@
+import { ethers } from 'ethers';
+import Web3 from 'web3';
+import { JsonRpcResponse } from 'web3-core-helpers';
+import {
+  GenericContainer,
+  StartedNetwork,
+  StartedTestContainer
+} from 'testcontainers';
+import { CHAIN_ID } from '../utils/constant';
+import { HttpProvider, WebsocketProvider } from 'web3-core';
+
+type MetaDNetwork = 'mainnet' | 'testnet';
+
+export interface StartOptions {
+  port?: number;
+  rpcPort?: number;
+  wsPort?: number;
+  timeout?: number;
+  spec?: number;
+}
+
+export class MetaDContainer {
+  static readonly PREFIX = 'metachain-testcontainers-';
+
+  static get image(): string {
+    if (process?.env?.METACHAIN_DOCKER_IMAGE !== undefined) {
+      return process.env.METACHAIN_DOCKER_IMAGE;
+    }
+    return 'ghcr.io/defich/metachain:af2e7d03b061352491d550c8923d1dfac4f65095';
+  }
+
+  static readonly MetaDPorts = {
+    mainnet: {
+      port: 30333,
+      rpcPort: 9933,
+      wsPort: 9944
+    },
+    testnet: {
+      port: 39333,
+      rpcPort: 19933,
+      wsPort: 19944
+    }
+  };
+
+  genericContainer: GenericContainer;
+  startedContainer?: StartedTestContainer;
+  startOptions?: StartOptions;
+  private network?: StartedNetwork;
+
+  web3!: Web3;
+  ethersjs!: ethers.providers.JsonRpcProvider;
+
+  constructor(
+    readonly metaDNetwork: MetaDNetwork = 'testnet',
+    readonly image: string = MetaDContainer.image,
+    readonly provider: string = 'http'
+  ) {
+    this.genericContainer = new GenericContainer(image);
+  }
+
+  protected getCmd(opts: StartOptions): string[] {
+    return [
+      '--execution=Native', // Faster execution compare to `Wasm`
+      '--no-telemetry', // disable connecting to substrate telemtry server
+      '--no-prometheus', // do not expose a Prometheus exporter endpoint
+      '--no-grandpa',
+      // TODO(canonbrother): set up chain spec exclusively for test
+      // '--chain= ./spec.json',
+      // `-l${FRONTIER_LOG}`,
+      `--port=${opts.port}`,
+      `--rpc-port=${opts.rpcPort}`,
+      `--ws-port=${opts.wsPort}`,
+      '--rpc-external',
+      '--ws-external',
+      '--sealing=manual',
+      '--force-authoring', // enable authoring even when offline
+      '--rpc-cors=all',
+      '--alice', // shortcut for `--name Alice --validator` with session keys for `Alice` added to keystore
+      '--tmp' // run a temporary node
+    ];
+  }
+
+  async start(startOptions: StartOptions = {}): Promise<void> {
+    this.startOptions = Object.assign(
+      MetaDContainer.MetaDPorts[this.metaDNetwork],
+      startOptions
+    );
+    const timeout = this.startOptions.timeout ?? 20000;
```

Comment:
```suggestion
    const timeout = this.startOptions.timeout ?? 100_000;
```
20_000 is a bit too short for my machine 😅